wiki/Client/Beatmap_editor/Song_setup/en.md

```diff
@@ -1,238 +1,121 @@
+By holding down `Shift`, it's possible to adjust the values with the 0.1 precision, as opposed to the standard step of 1.
 
 ### HP drain rate
 
-HP drain rate, commonly abbreviated as HP, determines the rate of health recovery from good hit scores and health punishment from bad hit scores. The value ranges from 0 to 10; higher HP values will give lower health recovery with higher health punishment.
-
-In osu! and osu!catch, this also affects the [Passive HP](/wiki/Client/Interface/Health_bar) (the constant rate at which HP decreases, except during the [Breaks](/wiki/Beatmap/Break).
+*Main article: [Health](/wiki/Beatmap/HP_drain_rate)*
 
-*Note: More HP is recovered with [Geki](/wiki/Gameplay/Judgement/Geki) and [Katu](/wiki/Gameplay/Judgement/Katu) hit scores, which can be given from the last hit object in a combo.*
+HP drain rate (HP) determines the amount of health recovery or damage from accurate or mistimed [hits](/wiki/Gameplay/Judgement). In osu! and osu!catch, it also affects the rate of [passive health drain](/wiki/Beatmap/HP_drain_rate). Higher values provide lower health recovery with higher punishment.
 
 ### Circle size
 
-![Overview of various circle sizes](img/Circle_sizes.png "Overview of various circle sizes")
-
-*Note: This option is visible in osu!taiko, but it does not do anything that affects gameplay.*
+*Main article: [Circle size](/wiki/Beatmap/Circle_size)*
 
-Circle size, commonly abbreviated as CS, exists only in osu! and osu!catch. CS determines the size of hit objects or, more specifically in osu!catch, determines the size of the catcher and fruit. The values range from 2 to 7; higher CS values will make hit objects smaller. Using the highest value will make the hit objects as big as the cursor.
-
-Most beatmaps use a value of 3 or 4, while a value of 5 or 6 is sometimes used and a value of 7 is almost never used. A value of 7 onwards is likely to be seen when using the [Hard Rock](/wiki/Gameplay/Game_modifier/Hard_Rock) mod.
-
-Setting the CS value to something less than 2 or greater than 7 can be done by manually editing the `.osu` file.
-
-#### Key Count
+![Overview of various circle sizes](img/Circle_sizes.png "Overview of various circle sizes")
 
-Key Count, also known as Keys, replaces the [circle size](#circle-size) metric for [osu!mania-specific](#mode-specific) beatmaps. Key Count is self-explanatory; it defines the key count that is used. The values range from 1 to 9 keys.
+Circle size (CS) determines the size of hit objects in osu! and osu!catch, while not affecting osu!taiko gameplay. Higher values will make hit objects smaller; while the editor limits CS to a 2 to 7 range, it's possible to overcome the limitation by manually editing the difficulty's [`.osu`](/wiki/Client/File_formats/osu_(file_format)) file.
```

Comment:
```suggestion
Circle size (CS) determines the size of hit objects in osu! and osu!catch. Higher values will make hit objects smaller. While the editor limits CS to a 2 to 7 range, it's possible to overcome the limitation by manually editing the difficulty's [`.osu`](/wiki/Client/File_formats/osu_(file_format)) file. Circle size has no effect in osu!taiko.
```
the semicolon sentence threw me off so much

Reply:
curious -- why is that?